openff/evaluator/backends/dask_kubernetes.py

```diff
@@ -0,0 +1,484 @@
+            assert isinstance(cpu_resources_per_worker, PodResources)
+            default_resources = cpu_resources_per_worker
+        else:
+            raise ValueError(
+                "Either gpu_resources_per_worker or cpu_resources_per_worker must be specified."
+            )
+
+        super().__init__(
+            default_resources._minimum_number_of_workers,
+            default_resources,
+        )
+
+        self._cluster_name = cluster_name
+        self._cluster_port = cluster_port
+        self._namespace = namespace
+        self._annotate_resources = annotate_resources
+        self._image = image
+        self._other_resources = other_resources
+        self._include_jupyter = include_jupyter
+        self._disable_nanny_process = disable_nanny_process
+        self._env = {}
+        if env is not None:
+            assert isinstance(env, dict)
+            self._env.update(env)
+
+        self._secrets = []
+        if secrets is not None:
+            assert isinstance(secrets, list)
+            for secret in secrets:
+                assert isinstance(secret, KubernetesSecret)
+                self._secrets.append(secret)
+
+        self._volumes = []
+        if volumes is not None:
+            assert isinstance(volumes, list)
+            for volume in volumes:
+                assert isinstance(volume, BaseKubernetesVolume)
+                self._volumes.append(volume)
+
+        # fail if there are no volumes -- we need volumes...
+        # unless we swap to S3?
+        if len(self._volumes) == 0:
+            raise ValueError("No volumes specified. We need at least a filesystem")
+
+        self._cluster_kwargs = {}
+        if cluster_kwargs is not None:
+            assert isinstance(cluster_kwargs, dict)
+            self._cluster_kwargs.update(cluster_kwargs)
+
+    def submit_task(self, function, *args, **kwargs):
+        from openff.evaluator.workflow.plugins import registered_workflow_protocols
+
+        key = kwargs.pop("key", None)
+
+        protocols_to_import = [
+            protocol_class.__module__ + "." + protocol_class.__qualname__
+            for protocol_class in registered_workflow_protocols.values()
+        ]
+
+        # look for simulation protocols
+        if self._annotate_resources:
+            resources = kwargs.get("resources", {})
+            if len(args) >= 2:
+                # schema is the second argument
+                # awful temporary terribad hack
+                schema_json = args[1]
+                if (
+                    '".allow_gpu_platforms": true' in schema_json
+                    or "energy_minimisation" in schema_json
+                ):
+                    resources["GPU"] = 0.5
+                    resources["notGPU"] = 0
+                else:
+                    resources["GPU"] = 0
+                    resources["notGPU"] = 1
+            kwargs["resources"] = resources
+            logger.info(f"Annotating resources: {resources}")
```

Comment:
This is a fairly hacky hard-coding that I did to pick up OpenMM minimization and simulation jobs. It's probably not the best solution for a number of reasons.

An alternative solution for identifying which protocols to divert to gpu/cpu workers include modifying somewhere upstream with direct access to the protocol (I think specifically https://github.com/openforcefield/openff-evaluator/blob/main/openff/evaluator/workflow/protocols.py#L1036-L1045) to either: a) pick up the Protocol by type, which could be hardcoded in a list, or b) add the `allow_gpu_platforms` attribute to the EnergyMinimisation protocol and look for that.

An alternative solution for choosing which workers to divert the protocol to is keeping the actual worker IDs in a dict somewhere and specifying them using the `workers` kwarg of `Client.submit` (https://distributed.dask.org/en/stable/api.html#distributed.Client.submit). This avoids hardcoding the very quickly-named custom resources of `GPU` and `notGPU`. 

However a pro of this custom resources solution is it's fairly isolated to just the DaskKubernetesBackend and shouldn't modify the existing behaviour with HPC workers.

Reply:
These are the [resources](https://distributed.dask.org/en/latest/resources.html) being used per worker -- they can be arbitrarily named but I think need to be numerical. I was setting `GPU=0.5` in the protocols and `GPU=1` on the worker to try to get two protocols to execute at once on the same GPU worker, but not sure it was more efficient.